pyspatial/raster.py

```diff
@@ -990,17 +1044,28 @@ def read_catalog(dataset_catalog_file):
     path = decoded["Path"]
     grid_size = decoded.get("GridSize", None)
     index = None
-    tile_regex = None
+    tile_structure = None
+    tms_z = None
 
     if "Index" in decoded:
```

Comment:
We should probably drop support for "Index" all together for now.  It's not really used anywhere, and needs more thought to be incorporated correctly.  I've deleted a bunch of code that used this since it wasn't behaving correctly, and didn't have more time to put towards it.